src/gui/processing/models/qgsmodelviewtoollink.cpp

```diff
@@ -0,0 +1,277 @@
+/***************************************************************************
+                             qgsmodelviewtoollink.cpp
+                             ------------------------------------
+    Date                 : January 2024
+    Copyright            : (C) 2024 Valentin Buira
+    Email                : valentin dot buira at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmodelviewtoollink.h"
+#include "moc_qgsmodelviewtoollink.cpp"
+#include "qgsprocessingmodelerparameterwidget.h"
+#include "qgsprocessingmodelalgorithm.h"
+#include "qgsgui.h"
+#include "qgsprocessingguiregistry.h"
+#include "qgsprocessingmodelchildalgorithm.h"
+#include "qgsmodelgraphicsscene.h"
+#include "qgsmodelviewmouseevent.h"
+#include "qgsmodelgraphicsview.h"
+#include <QScrollBar>
+
+QgsModelViewToolLink::QgsModelViewToolLink( QgsModelGraphicsView *view )
+  : QgsModelViewTool( view, tr( "Link Tool" ) )
+{
+  setCursor( Qt::PointingHandCursor );
+  mBezierRubberBand.reset( new QgsModelViewBezierRubberBand( view ) );
+
+  mBezierRubberBand->setBrush( QBrush( QColor( 0, 0, 0, 63 ) ) );
+  mBezierRubberBand->setPen( QPen( QBrush( QColor( 0, 0, 0, 100 ) ), 0, Qt::SolidLine ) );
+}
+
+void QgsModelViewToolLink::modelMoveEvent( QgsModelViewMouseEvent *event )
+{
+  mBezierRubberBand->update( event->modelPoint(), Qt::KeyboardModifiers() );
+
+  // we need to manually pass this event down to items we want it to go to -- QGraphicsScene doesn't propagate
+  QList<QGraphicsItem *> items = scene()->items( event->modelPoint() );
+
+  QgsModelDesignerSocketGraphicItem *socket = nullptr;
+  for ( QGraphicsItem *item : items )
+  {
+    if ( ( socket = dynamic_cast<QgsModelDesignerSocketGraphicItem *>( item ) ) )
+    {
+      // snap
+      if ( mFromSocket != socket && mFromSocket->edge() != socket->edge() )
+      {
+        socket->modelHoverEnterEvent( event );
+        QPointF rubberEndPos = socket->mapToScene( socket->getPosition() );
+        mBezierRubberBand->update( rubberEndPos, Qt::KeyboardModifiers() );
+
+        break;
+      }
+    }
+  }
+
+  if ( socket == nullptr && mLastHoveredSocket != nullptr && socket != mLastHoveredSocket )
```

Comment:
```suggestion
  if ( !socket && mLastHoveredSocket && socket != mLastHoveredSocket )
```

(see https://github.com/qgis/QGIS-Enhancement-Proposals/blob/master/qep-314-coding-style.md#3-language-features 3.3 )